packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -329,8 +369,24 @@ export class EngineReportingAgent<TContext = any> {
     handleLegacyOptions(this.options);
   }
 
+  public executableSchemaIdGenerator(schema: string | GraphQLSchema) {
+    // TODO: At somepoint fix this caching for managed federation so we treat it as an  LRU and gc old schemas.
+    const cachedId = this.executableSchemaToIdMap.get(schema);
```

Comment:
What's the point of making a map if we only want there to be one entry? We have a `// TODO` to turn a map that only has one entry into an LRU cache, and I'm not sure why.

Reply:
You can have a single cached value of 
```
interface SchemaAndExecutableId {
  executableSchemaId: string
  schema: string | GraphQLSchema
}
```

and if the schema isn't identical, then clear it instead.

Reply:
Switched to just caching the last value.